server/lib/routes/post-csp.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Route to report CSP Violations to metrics
+ */
+
+var DEFAULT_RATE_LIMIT = 90;
+
+module.exports = function (options) {
+  options = options || {};
+
+  /**
+   * Rate limit 'rateLimitPercentage' % of messages.
+   * @param rateLimitPercentage
+   * @returns {boolean}
+   */
+  function isRateLimited(rateLimitPercentage) {
+    var RATE_LIMIT = rateLimitPercentage;
+    // rate limit can be 0
+    if (typeof rateLimitPercentage === 'undefined') {
+      RATE_LIMIT = DEFAULT_RATE_LIMIT;
+    }
+    var min = 0;
+    var max = 100;
+
+    var rand = min + Math.floor(Math.random() * (max - min + 1));
+    return rand <= RATE_LIMIT;
+  }
+
+  return {
+    method: 'post',
+    path: '/_/csp-violation',
+    process: function (req, res) {
+      res.json({result: 'ok'});
+
+      // TODO: This is a temporary measure
+      // Not sure how many CSP errors we will get, for now we rate limit this.
+      // To avoid overflowing Heka logs rate limit the logging
+      if (isRateLimited(options.rateLimitPercentage)) {
```

Comment:
Using `isRateLimited` seems a little strange and is the reverse of how metrics are handled. Metrics uses [isSampledUser](https://github.com/mozilla/fxa-content-server/blob/327d67ded5ae1d738fbb1b5e4264038a2bed7c64/app/scripts/lib/app-start.js#L271) and a sampling percentage, whereas here we use a rate limiting percentage. 0% sample rate is no samples are let through, 0% rate limiting means all CSP reports are let through. For our own future sanity, we should use a consistent scheme in both places. I propose changing `options.rateLimitPercentage` to something like `options.cspReportSamplePercentage`